### PR TITLE
Upgrade to PostCSS 5.x

### LINKED ISSUE #202: Upgrade to postcss 5.x
Most plugins are ready but there is still some work to prepare.

### PR DESCRIPTION
**Pull request by jedmao**
Fixes #202 

These should be the proper version upgrades for plugins on the PostCSS 5.x path; though, a few PRs are still waiting to be merged in, so this build won't pass until they are merged and published.

### CODE REVIEW

**Comment by MoOx** [comment]
I just merged/published last requirements for updated plugins but I think we will have some issues with async plugins. cssnext default API is sync so we will run into issues. I restarted CI builds.

**Comment by jedmao** [comment]
Looks like there's an issue w/ the microtime package. I've commented on this thread: https://github.com/wadey/node-microtime/issues/34

**Comment by jedmao** [comment]
@MoOx it looks like AppVeyor has false positives. I'm also seeing some deprecation warnings, for which I've submitted PRs at https://github.com/iamvdo/pleeease-filters/pull/12 and https://github.com/postcss/postcss-browser-reporter/pull/20.

The microtime commit is in reference to https://github.com/wadey/node-microtime/issues/34#issuecomment-138442795.

Lastly, it looks like Travis is exposing [the issue you mentioned](https://github.com/cssnext/cssnext/pull/209#issuecomment-138435650).

```
1094cssnext encounters an error:
1095
1096Use process(css).then(cb) to work with async plugins
```

What's the next step here? Or should we get started on the `postcss-cssnext` plugin? If so, what do you need from me?

File: .travis.yml [hunk 1]
```diff
@@ -1,4 +1,12 @@
 sudo: false
+env:
+  - CXX=g++-4.8
+addons:
+  apt:
+    sources:
+    - ubuntu-toolchain-r-test
+    packages:
+    - g++-4.8
```
Comment (MoOx) [inline .travis.yml]: are those weird things required?

**Comment by MoOx** [comment]
@jedmao I just created a repo for postcss-cssnext https://github.com/cssnext/postcss-cssnext
You have been invited to cssnext org, feel free to make some PR.

Here is the idea: we should extract most of the code of this repo that can be and defined postcss-cssnext as a postcss plugin. Feel free to open some issues overthere so we can iterate quickly on this.
Travis has been setup for this repo, we might just need to push a .travis.yml ;)
Thanks for your involvement !

After that, we might work on a new release for cssnext based on this plugin that will bring a major breaking change in the current nodejs api.

Do not hesitate if you have any question.

File: .travis.yml (hunk 1 above)
Comment (jedmao) [inline .travis.yml]: According to https://github.com/wadey/node-microtime/issues/34#issuecomment-138529797, yes. I'd be happy to prove me wrong though. I don't like it any more than you do!

Comment (MoOx) [inline .travis.yml]: So feel free to add a comment in the travis yml about that, so future contributors might understand what is it about (or even you in the future :D)

Comment (jedmao) [inline .travis.yml]: I've added comments to the new microtime commit https://github.com/jedmao/cssnext/commit/dba072a1e332daea84535779abc9006bbc7732b2

**Comment by MoOx** [comment]
This won't happen. See #208.

**Pull request closed without merging**